### ISSUE TITLE
Unsigned request option

### ISSUE THREAD

**Issue by wjordan** [aws/aws-sdk-ruby#1149]
The AWS CLI has a [`--no-sign-request`](https://github.com/aws/aws-cli/commit/64822485f48d2d0f7916d957ae5fd9e4e68cd907) option which I've found useful for anonymously downloading S3 objects with `public-read` ACLs, and I'd like to see a similar feature added to the Ruby SDK.

Although it's currently possible to pass an S3 object's `#public_url` to [`open`](http://ruby-doc.org/stdlib-2.3.0/libdoc/open-uri/rdoc/OpenURI.html) or another Ruby HTTP client, I'd like to be able to leverage the AWS SDK's existing functionality for optimized downloading, file integrity verification and network-error retries, but without requiring credentials to be provided when not necessary.

**Comment by trevorrowe** [comment]
This is something I've been considering adding to the SDK recently. I'm still considering what the mechanism would be to allow for unauthenticated requests.

**Comment by awood45** [comment]
Added to feature request backlog.

**Comment by Techbrunch** [comment]
@awood45 is this still in the request backlog ?

**Comment by mullermp** [comment]
Reopening - deprecating usage of Feature Requests backlog markdown file.

**Comment by yashdave31** [comment]
Hey @mullermp, is this still in the feature backlog?

**Comment by mullermp** [comment]
Yes, but it's not prioritized. I think that there is a workaround however. The Clients support an undocumented `unsigned_operations` key. I can't guarantee the backwards compatibility for it going forward (like in a new major version), but you can do something like this:

```
Aws::S3::Client.new(unsigned_operations: [:get_object]).get_object(bucket: 'bucket', key: 'test')
```

In the future we are trying to do pluggable signers, and in a case like this, it would be an AnonymousSigner object but that is a long term goal.
